pkgs/shells/bash/5.nix

```diff
@@ -91,7 +91,7 @@ stdenv.mkDerivation rec {
 
   strictDeps = true;
   # Note: Bison is needed because the patches above modify parse.y.
-  depsBuildBuild = [ buildPackages.stdenv.cc ];
+  depsBuildBuild = [ stdenv ];
```

Comment:
Why this change? I've never seen depending on entire other stdenvs like that.

Reply:
If we just skip this, everything else looks good to me.

Reply:
> Why this change?

Because `!(buildPackages.stdenv.cc?__spliced)`.

See https://github.com/NixOS/nixpkgs/pull/264992 for the test case.

The reproducer requires two platforms which have the same triple but aren't `==`-equal to each other.  People are creating them on their own and getting burned by this.  Latest example:

- https://github.com/NixOS/nixpkgs/issues/265121

There aren't any in `lib.systems.examples`, so #264992 adds one in order to be able to answer your question "why this change".

Reply:
> Why this change?

Because: https://github.com/NixOS/nixpkgs/pull/264992#issuecomment-1791961285

Reply:
> If we just skip this, everything else looks good to me.

I moved it into

- https://github.com/NixOS/nixpkgs/pull/265162